src/auth/linking.ts

```diff
@@ -0,0 +1,242 @@
+import aes from "keystore-idb/lib/aes/index.js"
+import config from "keystore-idb/lib/config.js"
+import rsa from "keystore-idb/lib/rsa/index.js"
+import utils from "keystore-idb/lib/utils.js"
+import { KeyUse, SymmAlg, HashAlg, CharSize } from "keystore-idb/lib/types.js"
+import * as did from "../did/index.js"
+import * as ucan from "../ucan/index.js"
+import { publishOnChannel } from "./index.js"
+import * as auth from "./index.js"
+
+type LinkingRole = "CONSUMER" | "PRODUCER"
+
+type LinkingStep = "BROADCAST" | "NEGOTIATION" | "DELEGATION"
+
+type LinkingState = {
+  sessionKey: CryptoKey | null
+  temporaryRsaPair: CryptoKeyPair | null
+  role: LinkingRole | null
+  step: LinkingStep | null
+}
+
+const ls: LinkingState = {
+  sessionKey: null,
+  temporaryRsaPair: null,
+  role: null,
+  step: null
+}
+
+const resetLinkingState = () => {
+  ls.sessionKey = null
+  ls.temporaryRsaPair = null
+}
+
+const nextStep = () => {
```

Comment:
we're keeping a very simple state machine here - one for each "role" in the exchange (producer and consumer) and tracking both in our `LinkingState`